src/testdir/test_signals.vim

```diff
@@ -106,3 +106,47 @@ func Test_signal_INT()
 
   call StopVimInTerminal(buf)
 endfunc
+
+" Test a deadly signal.
+"
+" There are several deadly signals: SISEGV, SIBUS, SIGTERM...
+" Test uses signal SIGTERM as it does not create a core
+" dump file unlike SIGSEGV, SIGBUS, etc. See "man 7 signals.
+"
+" Vim should exit with a deadly signal and unsaved changes
+" should be recoverable from the swap file preserved as a
+" result of the deadly signal handler.
+func Test_deadly_signal_TERM()
+  if !HasSignal('TERM')
+    throw 'Skipped: TERM signal not supported'
+  endif
+  if !CanRunVimInTerminal()
+    throw 'Skipped: cannot run vim in terminal'
+  endif
+  let cmd = GetVimCommand()
+  if cmd =~ 'valgrind'
+    throw 'Skipped: cannot test signal TERM with valgrind'
+  endif
+
+  let buf = RunVimInTerminal('Xsig_TERM', {'rows': 6})
+  let pid_vim = term_getjob(buf)->job_info().process
+
+  call term_sendkeys(buf, ":call setline(1, 'foo')\n")
+  call WaitForAssert({-> assert_equal('foo', term_getline(buf, 1))})
+
+  call assert_false(filereadable('Xsig_TERM'))
+  exe 'silent !kill -s TERM '  .. pid_vim
+  call WaitForAssert({-> assert_equal('Vim: Caught deadly signal TERM', term_getline(buf, 1))})
+  call WaitForAssert({-> assert_match('Vim: preserving files\.\.\.$', term_getline(buf, 2))})
```

Comment:
One thing I noticed, is that when doing `kill -s TERM <pid_of_vim>`
vim outputs:
```
Vim: Caught deadly signal TERM
                              Vim: preserving files...
                                                      Vim: Finished.
```
Notice the odd spacing.
Shouldn't we do this:
```
diff --git a/src/misc1.c b/src/misc1.c
index 6686a3514..a370a82f8 100644
--- a/src/misc1.c
+++ b/src/misc1.c
@@ -2174,7 +2174,7 @@ preserve_exit(void)
     {
        if (buf->b_ml.ml_mfp != NULL && buf->b_ml.ml_mfp->mf_fname != NULL)
        {
-           OUT_STR("Vim: preserving files...\n");
+           OUT_STR("Vim: preserving files...\r\n");
            screen_start();         // don't know where cursor is now
            out_flush();
            ml_sync_all(FALSE, FALSE);  // preserve all swap files
@@ -2184,7 +2184,7 @@ preserve_exit(void)
 
     ml_close_all(FALSE);           // close all memfiles, without deleting
 
-    OUT_STR("Vim: Finished.\n");
+    OUT_STR("Vim: Finished.\r\n");
 
     getout(1);
 }
diff --git a/src/os_unix.c b/src/os_unix.c
index 8424b11a3..596fb0e90 100644
--- a/src/os_unix.c
+++ b/src/os_unix.c
@@ -1083,10 +1083,10 @@ deathtrap SIGDEFARG(sigarg)
 
     // No translation, it may call malloc().
 #ifdef SIGHASARG
-    sprintf((char *)IObuff, "Vim: Caught deadly signal %s\n",
+    sprintf((char *)IObuff, "Vim: Caught deadly signal %s\r\n",
                                                         signal_info[i].name);
 #else
-    sprintf((char *)IObuff, "Vim: Caught deadly signal\n");
+    sprintf((char *)IObuff, "Vim: Caught deadly signal\r\n");
 #endif
 
     // Preserve files and exit.  This sets the really_exiting flag to prevent
```

It would then output the nicer looking:
```
Vim: Caught deadly signal TERM
Vim: Finished.
Terminated
```